doc/index.md

```diff
@@ -46,16 +46,65 @@ file and the CREDITS file for the licensing and acknowledgements.
 --------------------------------------------
 The code is known to work on the following compilers:
 
-- clang 3.6.2
-- GCC 4.9.1
+- clang 3.6.2 and later
+- GCC 4.9.1 and later
 
 \section tutorial-quick-start Quick Start
 
 --------------------------------------------
-Range v3 is a generic library that augments the existing standard library with
-facilities for working with *ranges*. A range can be loosely thought of a pair
-of iterators, although they need not be implemented that way. Bundling begin/end
-iterators into a single object brings several benefits.
+
+## What This Library Provides
+
+Range v3 is a generic library that augments the existing standard library with facilities for working with *ranges*. This means that you do not have to fiddle with `begin` and `end` iterators directly, but pass something around, that represents both in one object. 
+
+For every STL algorithm this library provides a version that accepts a *range* instead of a pair of iterators. 
+
+All transforming and filtering algorithms are composable. This means applying an algorithm to a range yields a **lazily** evaluated representation (a so-called **view**) of the result, which can either be evaluated to obtain the results of the applied transformation or can be passed to another algorithm.  
+
+This avoids temporary results while allowing a declarative and easy-to-understand programming style. Generally this is possible without sacrificing performance.  
+
+Range v3 also provides in-place modifying algorithms e.g. for sorting. These are called **action**s.  
+
+Range v3 is a contribution to making C++ easier to write and reason about. With Range v3 one can write code this way: 
+
+~~~~~~~{.cpp}
+    std::vector<PolarCoordinate> coordinates;
+    
+    auto outsideOfReasonableRange =
+        (PolarCoordinate pc) { return pc.distance > 3.0; };
+
+    auto is_inf_or_nan =
+        [](PolarCoordinate pc) { 
+          return std::isinf(pc.distance) 
+            || std::isnan(pc.distance); 
+            };
+
+    auto filteredAndInCartesianCoordinates = coordinates
+        | remove_if(outsideOfReasonableRange)
+        | remove_if(is_inf_or_nan)
+        | transform(polarToCartesian)
+        | reverse;
```

Comment:
Please be explicit about namespaces here. `| view::remove_if(...)`, etc. And have a `using namespace ranges;` or something at the top of the example.